src/store/indexeddb-remote-worker.js

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import q from "q";
+import LocalIndexedDBStoreBackend from "./indexeddb-local-backend.js";
+
+/**
+ * This class lives in the webworker and drives a LocalIndexedDbStoreBackend
+ * controlled by messages from the main process.
+ */
+class IndexedDbStoreWorker {
+    constructor(postMessage) {
```

Comment:
Done, but the lack of window is intentional: it's a web worker, there is no window, the variables live in the scope the worker runs in.